From 22b3efbe9b6ba4d0d35dba64c27a8ebd920140d3 Mon Sep 17 00:00:00 2001
From: Rob Clark <robclark@freedesktop.org>
Date: Mon, 18 Jun 2018 13:05:01 -0400
Subject: [PATCH 06/14] freedreno: add user ptr to fd_ringbuffer

Something for users of fd_ringbuffer to use as they see fit.  (For now,
just so mesa can add some debugging state.)

Signed-off-by: Rob Clark <robclark@freedesktop.org>
---
 freedreno/freedreno_ringbuffer.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/freedreno/freedreno_ringbuffer.h b/freedreno/freedreno_ringbuffer.h
index c501fba..0a3d6f7 100644
--- a/freedreno/freedreno_ringbuffer.h
+++ b/freedreno/freedreno_ringbuffer.h
@@ -47,6 +47,11 @@ struct fd_ringbuffer {
 	const struct fd_ringbuffer_funcs *funcs;
 	uint32_t last_timestamp;
 	struct fd_ringbuffer *parent;
+
+	/* for users of fd_ringbuffer to store their own private per-
+	 * ringbuffer data
+	 */
+	void *user;
 };
 
 struct fd_ringbuffer * fd_ringbuffer_new(struct fd_pipe *pipe,
-- 
1.9.1

